hooks/cache_location.py

```diff
@@ -76,36 +76,28 @@ def _get_cache_root(self, project_id, pipeline_configuration_id):
         and inside this base level hook. This method calculates the cache root
         for the current project and configuration. In the default implementation,
         all the different types of cache data resides below a common root point. 
-        
+
         :param project_id: The shotgun id of the project to store caches for
         :param pipeline_configuration_id: The shotgun pipeline config id to store caches for
         :returns: The calculated location for the cache root
         """
-        
         # the default implementation will place things in the following locations:
         # macosx: ~/Library/Caches/Shotgun/SITE_NAME/project_xxx/config_yyy
         # windows: $APPDATA/Shotgun/SITE_NAME/project_xxx/config_yyy
         # linux: ~/.shotgun/SITE_NAME/project_xxx/config_yyy
-        
+
         # first establish the root location
+
         tk = self.parent
-        if sys.platform == "darwin":
-            root = os.path.expanduser("~/Library/Caches/Shotgun")
-        elif sys.platform == "win32":
-            root = os.path.join(os.environ["APPDATA"], "Shotgun")
-        elif sys.platform.startswith("linux"):
-            root = os.path.expanduser("~/.shotgun")
 
-        # get site only; https://www.foo.com:8080 -> www.foo.com
-        base_url = urlparse.urlparse(tk.shotgun.base_url)[1].split(":")[0]
-        
+        root = tk.get_site_cache_root()
```

Comment:
Wondering about the name `get_site_cache_root()` - I think we should emphasize that it is a local place that is organized per user! Could it be a property called `tk.local_cache_location` or `tk.local_site_cache_location` or something? This would be consistent with the `bundle.cache_location` property.

On more thought - is it strange that this is a method on the `tk` instance but it returns something site specific? I would sort of intuitively think that two tk instances belong to two different projects on the same site would return two different cache locations - but they return the same?